runtime-filter.md

```diff
@@ -0,0 +1,257 @@
+- Build side: one side of hash join used to build a hash table. In this document, the right table of Join is called the build side by default.
+- Probe side: one side of hash join used to continuously match the hash table. In this document, the left table of Join is called the probe side by default.
+- Filter: also known as predicate, which means the filter condition in this document.
+
+## Working principles of Runtime Filter
+
+Hash join performs the join operation by building a hash table on the right table and continuously probing the hash table on the left table. If some join key values cannot hit the hash table during the probing process, it means that these data do not exist in the right table and will not appear in the final join result. Therefore, if TiDB can **filter out these join key data in advance** during scanning, it will reduce the scanning time and network overhead, thereby greatly improving the join efficiency.
+
+Runtime Filter is a **dynamic predicate** generated during the query planning phase. This predicate has the same function as other predicates in the TiDB Selection operator. These predicates are all applied to the Table Scan operation to filter out rows that do not match the predicate conditions. The only difference is that the parameter values in Runtime Filter come from the results generated during the Hash Join build process.
+
+### Example
+
+Assume that there is a Join query between the `store_sales` table and the `date_dim` table, and the Join method is Hash Join. `store_sales` is a fact table that mainly stores the sales data of stores, and the number of rows is 1 million. `date_dim` is a time dimension table that mainly stores date information. Currently, you want to query the sales data of year 2001, so the number of rows of `date_dim` table involved in the Join operation is 365.
+
+```sql
+SELECT * FROM store_sales, date_dim
+  WHERE ss_date_sk = d_date_sk
+    AND d_year = 2001;
+```
+
+The execution method of Hash Join is usually as follows:
+
+```
+                 +-------------------+
+                 | PhysicalHashJoin  |
+        +------->|                   |<------+
+        |        +-------------------+       |
+        |                                    |
+        |                                    |
+  100w  |                                    | 365
+        |                                    |
+        |                                    |
++-------+-------+                   +--------+-------+
+| TableFullScan |                   | TableFullScan  |
+|  store_sales  |                   |    date_dim    |
++---------------+                   +----------------+
+```
+
+*(The above figure omits the exchange node and other nodes.)*
+
+Runtime Filter 的执行方式如下：
+
+The execution process of Runtime Filter is as follows:
+
+1. Scan the data of `date_dim`, and PhysicalHashJoin calculates a filter condition based on the `date_dim` data, such as `date_dim in (2001/01/01~2001/12/31)`.
```

Comment:
Add a comment to https://github.com/pingcap/docs-cn/pull/14517#discussion_r1280477767